engine/execution/computation/computer/computer_test.go

```diff
@@ -362,6 +363,48 @@ func TestBlockExecutor_ExecuteBlock(t *testing.T) {
 		assertEventHashesMatch(t, collectionCount+1, result)
 	})
 
+	t.Run("system transaction does not read memory limit from state", func(t *testing.T) {
+
+		weighted.DefaultMemoryWeights = weighted.ExecutionMemoryWeights{
+			0: 1, // single weight set to 1
+		}
+		execCtx := fvm.NewContext(
+			zerolog.Nop(),
+			fvm.WithMemoryLimit(10), // the context memory limit is set to 10
+		)
```

Comment:
this is a nice test, but we might add a test for the AllowContextOverrideByExecutionState functionality to the FVM as well.